Source/BlockchainService/BlockchainService.swift

```diff
@@ -5,4 +5,6 @@ protocol BlockchainService {
     func configuration(ofNodeAtURL url: URL, completion: ((Result<[Node]>) -> Void)?)
     func submit(hashes: [String], forNumberOfNodes: UInt, completion: ((Result<[NodeHash]>) -> Void)?)
     func submit(hashes: [String], toNodeURLs urls: [NodeURI], completion: ((Result<[NodeHash]>) -> Void)?)
+    
+    func proof(forHashId: HashIdNode, completion: @escaping (Result<Proof>) -> Void)
```

Comment:
We can not support only `HashIdNode` proof retrieval, because we need to know for what public uri was the hash uploaded to. Therefore the API signature should only support asking for `NodeHash` type, because it has all the node uri-s encapsulated there.